### PR TITLE
Enable nullability in DesignerFrame and clean it up

### PR DESCRIPTION
**Pull request by halgab**
## Proposed changes

- Replace `ArrayList`s with `List<T>`s in `DesignerFrame` (contributes to #8140)
- Enable nullability in `DesignerFrame`
- Clean up `DesignerFrame` and one call site

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9966)

### CODE REVIEW

**Comment by LeafShi1** [comment]
LGTM！

**Review by LeafShi1 — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs [hunk 1]
```diff
@@ -132,17 +115,14 @@ public void Initialize(Control view)
     protected override void OnGotFocus(EventArgs e)
     {
         ForceDesignerRedraw(true);
-        ISelectionService selSvc = (ISelectionService)_designerSite.GetService(typeof(ISelectionService));
-        if (selSvc is not null)
+        ISelectionService? selSvc = _designerSite.GetService<ISelectionService>();
+        if (selSvc?.PrimarySelection is Control { IsDisposed: false } ctrl)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs:119]: we need consistency in adapting this syntax standard. Will get this reviewed collectively and make decision across the PRs.

Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs:119]: I'm happy to rollback such changes if it's a contentious issue. Here it doesn't add much to the table anyway.

Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs:119]: Not yet. Will update here if needed. We just need consistency in adapting these.

**Pull request merged**